### PR TITLE
added Liquibase support; implemented the soft delete concept.

### PR DESCRIPTION
**Pull request by misha-levko**
Added Endpoints to BookController

### CODE REVIEW

File: book-store-spring/src/main/java/mate/academy/bookstorespring/service/BookServiceImpl.java [hunk 1]
```diff
@@ -43,17 +43,13 @@ public void deleteById(Long id) {
...
-        bookUpdate.setTitle(bookDto.getTitle());
-        bookUpdate.setAuthor(bookDto.getAuthor());
-        bookUpdate.setPrice(bookDto.getPrice());
-        bookUpdate.setIsbn(bookDto.getIsbn());
-        bookUpdate.setDescription(bookDto.getDescription());
-        bookUpdate.setCoverImage(bookDto.getCoverImage());
-
-        return bookMapper.toDto(bookRepository.save(bookUpdate));
+    public BookDto updateBookById(CreateBookRequestDto bookRequestDto, Long id) {
+        bookRepository.findById(id).orElseThrow(() ->
```
Comment (Elena-Bruyako) [inline book-store-spring/src/main/java/mate/academy/bookstorespring/service/BookServiceImpl.java]:
```suggestion
        Book book = bookRepository.findById(id).orElseThrow(() ->
```

File: book-store-spring/src/main/java/mate/academy/bookstorespring/service/BookServiceImpl.java [hunk 2]
```diff
@@ -43,17 +43,13 @@ public void deleteById(Long id) {
...
+                new EntityNotFoundException(
+                        "Can't find book by id: " + id));
+        Book book = bookMapper.toModel(bookRequestDto);
+        book.setId(id);
+        Book updatedBook = bookRepository.save(book);
```
Comment (Elena-Bruyako) [inline book-store-spring/src/main/java/mate/academy/bookstorespring/service/BookServiceImpl.java]:
```suggestion
        bookMapper.updateBookFromDto(bookRequestDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

File: book-store-spring/src/main/java/mate/academy/bookstorespring/repository/BookRepository.java [hunk 3]
```diff
@@ -5,6 +5,5 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
-
     Optional<Book> getBookById(Long id);
```
Comment (Elena-Bruyako) [inline book-store-spring/src/main/java/mate/academy/bookstorespring/repository/BookRepository.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Task not completed

You need to implement: 
Add Liquibase support
Change spring.jpa.hibernate.ddl-auto property value to validate
Implement soft delete concept for the Book entity

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
didn't find liquibase changelogs in your implementation

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**